native-engine/datafusion-ext-plans/src/agg/agg_buf.rs

```diff
@@ -633,6 +700,36 @@ impl AggDynValue for AggDynSet {
     }
 }
 
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub enum OptimizeSet {
+    Empty,
+    LitteVec(SmallVec<[ScalarValue; 4]>),
+    Set(HashSet<ScalarValue>),
+}
+
+impl Default for OptimizeSet {
+    fn default() -> Self {
+        OptimizeSet::Empty
```

Comment:
can we use `OptimizedSet::SmallVec(SmallVec::default())` as default value? so that `Empty` is no longer used.